src/CadViewer.tsx

```diff
@@ -1,112 +1,144 @@
-      if (!internalCircuitJson) return [5, 5, 5] as const
-      try {
-        const board = su(internalCircuitJson as any).pcb_board.list()[0]
-        if (!board) return [5, 5, 5] as const
-        const { width, height } = board
+  const handleMenuClick = (newEngine: "jscad" | "manifold") => {
+    setEngine(newEngine)
+    setMenuVisible(false)
+  }
 
-        if (!width && !height) {
-          return [5, 5, 5] as const
-        }
+  const handleClickAway = useCallback((e: MouseEvent) => {
+    const target = e.target as Node
+    if (!menuRef.current || !menuRef.current.contains(target)) {
+      setMenuVisible(false)
+    }
+  }, [])
 
-        const minCameraDistance = 5
-        const adjustedBoardWidth = Math.max(width, minCameraDistance)
-        const adjustedBoardHeight = Math.max(height, minCameraDistance)
-        const largestDim = Math.max(adjustedBoardWidth, adjustedBoardHeight)
-        return [largestDim / 2, largestDim / 2, largestDim] as const
-      } catch (e) {
-        console.error(e)
-        return [5, 5, 5] as const
-      }
-    }, [internalCircuitJson])
+  useEffect(() => {
+    if (menuVisible) {
+      document.addEventListener("mousedown", handleClickAway)
+      return () => document.removeEventListener("mousedown", handleClickAway)
+    }
+  }, [menuVisible, handleClickAway])
 
-    // Use the state `boardGeom` which starts simplified and gets updated
-    const { stls: boardStls, loading } = useStlsFromGeom(boardGeom)
+  useEffect(() => {
+    const stored = window.localStorage.getItem("cadViewerEngine")
+    if (stored === "jscad" || stored === "manifold") {
+      setEngine(stored)
+    }
+  }, [])
 
-    if (!internalCircuitJson) return null
+  useEffect(() => {
+    window.localStorage.setItem("cadViewerEngine", engine)
+  }, [engine])
 
-    const cad_components = su(internalCircuitJson).cad_component.list()
+  const viewerKey = props.circuitJson
+    ? JSON.stringify(props.circuitJson)
+    : undefined
 
-    return (
-      <CadViewerContainer
-        ref={ref}
-        autoRotateDisabled={autoRotateDisabled}
-        initialCameraPosition={initialCameraPosition}
-        clickToInteractEnabled={clickToInteractEnabled}
+  return (
+    <div
+      key={viewerKey}
+      ref={containerRef}
+      style={{ width: "100%", height: "100%", position: "relative" }}
+      onContextMenu={handleContextMenu}
+    >
+      {engine === "jscad" ? (
+        <CadViewerJscad {...props} />
+      ) : (
+        <CadViewerManifold {...props} />
+      )}
+      <div
+        style={{
+          position: "absolute",
+          right: 8,
+          top: 8,
+          background: "#222",
+          color: "#fff",
+          padding: "2px 8px",
+          borderRadius: 4,
+          fontSize: 12,
+          opacity: 0.7,
+          userSelect: "none",
+        }}
+        onClick={() => {
+          if ("ontouchstart" in window) {
+            setEngine(engine === "jscad" ? "manifold" : "jscad")
```

Comment:
Did you test this? Will this not literally change the 3d viewer on every press making touch gestures on the the model not possible?